AdyenCard/Components/GiftCardComponent/GiftCardComponent.swift

```diff
@@ -0,0 +1,312 @@
+        item.buttonSelectionHandler = { [weak self] in
+            self?.didSelectSubmitButton()
+        }
+        return item
+    }()
+
+    // MARK: - Localizable Protocol
+
+    /// :nodoc:
+    public var localizationParameters: LocalizationParameters?
+
+    // MARK: - Loading Component Protocol
+
+    /// :nodoc:
+    public func stopLoading() {
+        button.showsActivityIndicator = false
+        viewController.view.isUserInteractionEnabled = true
+    }
+
+    /// :nodoc:
+    internal func startLoading() {
+        button.showsActivityIndicator = true
+        viewController.view.isUserInteractionEnabled = false
+    }
+
+    private func showError(message: String) {
+        errorItem.message = message
+        errorItem.isHidden.wrappedValue = false
+    }
+
+    private func hideError() {
+        errorItem.message = nil
+        errorItem.isHidden.wrappedValue = true
+    }
+
+    // MARK: - Submitting the form
+
+    internal func didSelectSubmitButton() {
+        guard formViewController.validate() else {
+            return
+        }
+
+        startLoading()
+
+        let data = createPaymentData()
+
+        partialPaymentDelegate?.checkBalance(data, from: self) { [weak self] result in
+            guard let self = self else { return }
+            result
+                .mapError(Error.otherError)
+                .flatMap(self.check(balance:))
+                .flatMap { isBalanceEnough in isBalanceEnough ? self.onReadyToPayFullAmount() : self.requestOrder() }
+                .handleError { self.handle(error: $0) }
+                .handleSuccess { /* Do nothing.*/ }
+        }
+    }
+
+    private func handle(error: Swift.Error) {
+        defer {
+            stopLoading()
+        }
+        if case let Error.otherError(internalError) = error {
+            delegate?.didFail(with: internalError, from: self)
+        }
+        guard let error = error as? BalanceValidator.Error else {
+            showError(message: "An unknown error occurred")
+            return
+        }
+        switch error {
+        case .unexpectedCurrencyCode:
+            showError(message: "Gift cards are only valid in the currency they were issued in")
+        case .zeroBalance:
+            showError(message: "This gift card has zero balance")
+        }
+    }
+
+    private func check(balance: Balance) -> Result<Bool, Swift.Error> {
+        guard let payment = payment else {
+            AdyenAssertion.assert(message: Error.invalidPayment.localizedDescription)
+            return .failure(Error.invalidPayment)
+        }
+        do {
+            return .success(try BalanceValidator().validate(balance: balance, toPay: payment.amount))
+        } catch {
+            return .failure(error)
+        }
+    }
+
+    private func onReadyToPayFullAmount() -> Result<Void, Swift.Error> {
+        AdyenAssertion.assert(message: "readyToSubmitComponentDelegate is nil",
```

Comment:
This will fail if not Drop-In .. 

Maybe `assert ` should do the other way around?
It should assert that condition is met. Otherwise it should be called `failOn`

Reply:
flipped the condition